### PR TITLE
Fix lints, create certificate from SSH wire format

### PR DESCRIPTION
**Pull request by 1Dragoon**
Mainly needed the ability to receive certificates as presented by the ssh client over the SSH wire protocol, but also fixed several lints that were marked as forbidden

### CODE REVIEW

File: src/ssh/cert.rs [hunk 1]
```diff
@@ -160,25 +160,30 @@ impl Certificate {
...
+    /// Reads an SSH certificate from a given byte sequence.
+    ///
+    /// The byte sequence is expected to be the base64 decoded body of the SSH certificate.
+    ///
+    pub fn from_bytes(data: &[u8]) -> Result<Certificate> {
+        let mut reader = Reader::new(&data);
 
         // Validate key types before reading the rest of the data
-        let kt_from_reader = reader.read_string()?;
-        if kt_name != kt_from_reader {
```
Comment (obelisk) [inline src/ssh/cert.rs:180]: I'm a little concerned this check appears to have been removed. I need to think more about if this is ok but I seem to remember adding this to catch certificates that do not state the correct type in their preamble

**Comment by obelisk** [comment]
Can you also add some tests for this new function?

**Comment by 1Dragoon** [comment]
> Can you also add some tests for this new function?

Hmm don't the existing tests effectively cover it since they already use the from_string() method, which itself makes complete use of the new method?

**Comment by obelisk** [comment]
Back from vacation and looking at this again. Again still thinking about that removed check

**Comment by 1Dragoon** [comment]
Doh just realized I have to "submit" the inline comments; merely posting them isn't enough

**Comment by obelisk** [comment]
Man I am bad at comments. Ok, I think I'm good to merge this. Going to think about it for another day, then hopefully merge it in. I have another large number of fixes I want to merge in so this has to get in before that as my incentive.

**Pull request merged**